src/ngx_http_lua_initworkerby.c

```diff
@@ -40,6 +40,25 @@ ngx_http_lua_init_worker(ngx_cycle_t *cycle)
 
     lmcf = ngx_http_cycle_get_module_main_conf(cycle, ngx_http_lua_module);
 
+    if (lmcf != NULL && lmcf->lua != NULL) {
```

Comment:
How about the following instead?

```C
    if (lmcf == NULL || lmcf->lua == NULL) {
        return NGX_OK;
    }

    /* lmcf != NULL && lmcf->lua != NULL */

    /* disable init_worker_by_lua* and destroy lua VM in cache processes */
    if (ngx_process == NGX_PROCESS_HELPER) {
#if defined(HAVE_PRIVILEGED_PROCESS_PATCH) && !NGX_WIN32
        if (!ngx_is_privileged_agent) {
#endif

            ngx_log_debug2(NGX_LOG_DEBUG_HTTP, ngx_cycle->log, 0,
                           "lua close the global Lua VM %p in "
                           "cache process %P", lmcf->lua, ngx_pid);
            lmcf->vm_cleanup->handler(lmcf->vm_cleanup->data);
            lmcf->vm_cleanup->handler = NULL;
            return NGX_OK;
#if defined(HAVE_PRIVILEGED_PROCESS_PATCH) && !NGX_WIN32
        }
#endif
    }

    if (lmcf->init_worker_handler == NULL) {
        return NGX_OK;
    }
```

This way we can save 2 duplicate checks.

Reply:
@agentzh
Good point!